src/SurfaceFluxes.jl

```diff
@@ -54,417 +57,817 @@ Surface flux conditions, returned from `surface_conditions`.
 
 $(DSE.FIELDS)
 """
-struct SurfaceFluxConditions{FT, VFT}
+struct SurfaceFluxConditions{FT}
     L_MO::FT
-    wθ_flux_star::FT
-    flux::VFT
-    x_star::VFT
-    C_exchange::VFT
+    shf::FT
+    lhf::FT
+    ρτxz::FT
+    ρτyz::FT
+    ustar::FT
+    Cd::FT
+    Ch::FT
 end
 
 function Base.show(io::IO, sfc::SurfaceFluxConditions)
     println(io, "----------------------- SurfaceFluxConditions")
-    println(io, "L_MO           = ", sfc.L_MO)
-    println(io, "wθ_flux_star = ", sfc.wθ_flux_star)
-    println(io, "flux           = ", sfc.flux)
-    println(io, "x_star         = ", sfc.x_star)
-    println(io, "C_exchange     = ", sfc.C_exchange)
+    println(io, "L_MO                   = ", sfc.L_MO)
+    println(io, "Sensible Heat Flux     = ", sfc.shf)
+    println(io, "Latent Heat Flux       = ", sfc.lhf)
+    println(io, "Friction velocity u⋆   = ", sfc.ustar)
+    println(io, "C_drag                 = ", sfc.Cd)
+    println(io, "C_heat                 = ", sfc.Ch)
     println(io, "-----------------------")
 end
 
-function surface_fluxes_f!(F, x, nt)
-    param_set = nt.param_set
-    wθ_flux_star = nt.wθ_flux_star
-    z_in = nt.z_in
-    z_0 = length(nt.z_0) > 1 ? Tuple(nt.z_0) : nt.z_0
-    x_in = Tuple(nt.x_in)
-    x_s = Tuple(nt.x_s)
-    n_vars = nt.n_vars
-    scheme = nt.scheme
-    universal_func = nt.universal_func
-    θ_scale = nt.θ_scale
+abstract type AbstractSurfaceLocation end
 
-    x_tup = Tuple(x)
+abstract type AbstractSurfaceConditions end
 
-    u_star, θ_star = x_tup[2], x_tup[3]
-    if wθ_flux_star == nothing
-        wθ_surf_flux = -u_star * θ_star
-    else
-        wθ_surf_flux = wθ_flux_star
-    end
-    L_MO = monin_obukhov_length(param_set, u_star, θ_scale, wθ_surf_flux)
-    uf = universal_func(param_set, L_MO)
-    F_nt = ntuple(Val(n_vars + 1)) do i
-        if i == 1
-            F_i =
-                x_tup[1] - monin_obukhov_length(
-                    param_set,
-                    u_star,
-                    θ_scale,
-                    wθ_surf_flux,
-                )
-        else
-            ϕ = x_tup[i]
-            transport = i - 1 == 1 ? UF.MomentumTransport() : UF.HeatTransport()
-            F_i =
-                ϕ - compute_physical_scale(
-                    uf,
-                    z_in,
-                    length(nt.z_0) > 1 ? z_0[i - 1] : z_0,
-                    x_in[i - 1],
-                    x_s[i - 1],
-                    transport,
-                    scheme,
-                )
-        end
-        F_i
-    end
-    F .= F_nt
+"""
+    Fluxes{VI, VS, AUF, FT}
+
+Input container, given surface state variables, latent and sensible heat fluxes.
+
+# Fields
+
+$(DSE.FIELDS)
+"""
+struct Fluxes{VI, VS, AUF, FT} <: AbstractSurfaceConditions
+    state_in::VI
+    state_sfc::VS
+    shf::FT
+    lhf::FT
+    uf::AUF
+    z0m::FT
+    z0b::FT
+    L_MO_init::FT
+    gustiness::FT
+end
+function GivenFluxes(
```

Comment:
Any reason why you use a different name for the constructor (i.e. `GivenFluxes` instead of `Fluxes`)?

Reply:
You can still do that without the constructor I think. But Charlie already resolved this.